src/routes/transactions/transactions.service.ts

```diff
@@ -218,6 +224,65 @@ export class TransactionsService {
     };
   }
 
+  private getAdjustedPaginationForHistory(
+    paginationData: PaginationData,
+  ): PaginationData {
+    if (paginationData.offset > 0) {
+      return new PaginationData(
+        paginationData.limit + 1,
+        paginationData.offset - 1,
+      );
+    }
+    return paginationData;
+  }
+
+  async getTransactionHistory(
+    chainId: string,
+    routeUrl: Readonly<URL>,
+    safeAddress: string,
+    paginationData: PaginationData,
+    timezoneOffset?: string,
+  ): Promise<Partial<TransactionItemPage>> {
+    const paginationDataAdjusted =
+      this.getAdjustedPaginationForHistory(paginationData);
+    const domainTransactions = await this.safeRepository.getTransactionHistory(
+      chainId,
+      safeAddress,
+      paginationDataAdjusted?.limit,
+      paginationDataAdjusted?.offset,
```

Comment:
Nitpick: these values can't be null/undefined anymore, I think the optional-chaining is not required here.

Reply:
Done 21343d26f082811dfac619628436595613a509eb